### ISSUE TITLE
JSON configurations

### ISSUE THREAD

**Issue by markNZed** [rlaffers/eslint-plugin-xstate#21]
There is a significant overlap between the static configuration of an Xstate machine and a JSON object (assuming the use of named actions/guards). Would it be possible for eslint-plugin-xstate to lint a JSON object that defines an XState machine?

**Comment by rlaffers** [comment]
Hi,
isn't this a duplicate of https://github.com/rlaffers/eslint-plugin-xstate/issues/20 ?
If you put the `/*eslint-plugin-xstate-include */` comment into your file with JSON object, it should be linted.

**Comment by markNZed** [comment]
Comments are not legal syntax in JSON files so I did not think to try. But it does not seem to work. I think ESLint does not run on JSON out of the box but I do not know much about this. I see there is https://github.com/azeemba/eslint-plugin-json Maybe there could be a setting to force plugin-xstate to run on a file name pattern (e.g. all JSON under a particular directory) ?

**Comment by rlaffers** [comment]
I see, you would like to lint JSON file (which is not the same as a JSON object). Linting JSON is not possible with ESLint as all - it is for linting Javascript only. I suggest you put your JSON object into a JS file and assign it to a variable to make the syntax valid.
```javascript
const config = {
  // your JSON
}
```

**Comment by markNZed** [comment]
Unfortunately that means other languages can't read the JSON without parsing Javascript. It seems https://github.com/azeemba/eslint-plugin-json does make it possible to run ESLint on JSON files but I doubt many people are interested in this functionality. In the bigger picture XState would ideally not be tied into using Javascript  (JSON configuration seems essential for this).

**Comment by rlaffers** [comment]
Yeah, XState is for Javascript only, at least at the moment. Thus there is not much need to read and lint XState configuration in other languages.

**Comment by markNZed** [comment]
I was discussing on their Discord and one of the team  replied "Xstate is in fact not tied to javascript at all other than having the typescript interpreter by default. There have been attempts to create statechart/xstate interpreters in other languages such as Python, .Net, etc." but I agree that given there are no other viable interpreters, Xstate is tied into Javsacript for now.
